expression/builtin_miscellaneous.go

```diff
@@ -204,7 +204,56 @@ type builtinInetAtonSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_inet-aton
 func (b *builtinInetAtonSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("INET_ATON")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	arg := args[0]
+	if arg.IsNull() {
+		return d, nil
+	}
+	s, err := arg.ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	// ip address should not end with '.'.
+	if len(s) == 0 || s[len(s)-1] == '.' {
+		return d, nil
+	}
+
+	var (
+		byteResult, result uint64
+		dotCount           int
+	)
+	for _, c := range s {
+		if c >= '0' && c <= '9' {
+			digit := uint64(c - '0')
+			byteResult = byteResult*10 + digit
+			if byteResult > 255 {
+				return d, nil
+			}
```

Comment:
After parsed numbers into []int, we can still fill zero in the slice if count is less than 4.